### ISSUE TITLE
No Monte Carlo Estimation During Test time?

### ISSUE THREAD

**Issue by allenanie** [julian121266/RecurrentHighwayNetworks#10]
Hi,

Just out of curiosity since I did see evaluate_mc function in the training file, but it was never actually used and the paper never talked about MC evaluation...my question is: did you guys compare the MC test result and non-MC test result? Is the non-MC test result better?

**Comment by jzilly** [comment]
Hi windweller,

We were originally curious about the MC evaluation described by Yarin Gal (https://github.com/yaringal/BayesianRNN) in his paper (https://nips.cc/Conferences/2016/Schedule?showEvent=7250). Essentially MC dropout runs the network M times on the test set **with dropout on**. These outputs of these runs are then averaged. 

From our experience this yields better results than simply turning off dropout at test time. Unfortunately it takes considerably longer to evaluate since the test set is evaluated up to 1000 times. 

In summary: The MC result is better in terms of performance, but computationally very expensive. This is why we ended up not pursuing this type of evaluation further.
